src/run.ts

```diff
@@ -160,24 +164,54 @@ export async function run(opts: RunOpts): Promise<void> {
                 .map(d => d.environment.id)
                 .filter((d): d is number => !!d),
               comment: `Superseded by workflow run ${current_run.html_url}`
-            })
+            });
           } else {
             await octokit.rest.actions.cancelWorkflowRun({
               owner,
               repo,
               run_id: wf.id
-            })
+            });
           }
         }
+        // Tag successful pull requests
+        if (opts.tagSuperseededWorkflowsWith !== "" && deployments.data.length > 0 ) {
+          if (wf.head_commit) {
+            const response = await octokit.rest.repos.listPullRequestsAssociatedWithCommit({
+              owner,
+              repo,
+              commit_sha: wf.head_commit.id
+            });
+        
+            if (response.data.length > 0) {
+              // Add a comment to the corresponding pull request
+              const pullRequest = response.data[0]
+              if (pullRequest) {
+                if (!opts.dryRun) {
+                  core.info(`Tagging pull request ${pullRequest.url} with ${opts.tagSuperseededWorkflowsWith}`);
+                  await octokit.rest.pulls.update({
+                    owner,
+                    repo,
+                    pull_number: pullRequest.number,
+                    labels: [opts.tagSuperseededWorkflowsWith],
+                  });
+                } else {
+                  core.info(`Dry run: tagged pull request ${pullRequest.url} with ${opts.tagSuperseededWorkflowsWith}`);
+                }
+              } else {
+                core.info(`No pull request found for commit ${wf.head_commit.id}`);
+              }
+            }
+          }
+        } 
       }
     }
-    core.info(`Completed cancellations for ${current_run.id}...`)
-    return
+    core.info(`Completed cancellations for ${current_run.id}...`);
```

Comment:
Unfortunately I think the way prettier is configured by the github actions repo template, it will drop the `;` at the end. If you run `npm run format`, it will remove these semicolons.